
\section{Experiment Framework}

In this case genome searching embarrassingly parallel. We want to compare viral genomes to ours, so naturally each virus comparison doesnâ€™t care about the state of the other comparisons are doing. In this sense we can break up our job into many sub jobs and run them independently on several machines. The experiment environment uses the Geni Experiment Engine to provide three nodes for computation \cite{Berman20145}. The nodes are controlled through a master using the python Fabric module. Fabric essentially allows shell commands to be run on remote machines through a python interface, and makes managing remote machines very simple. I used the SageFS distributed filesystem to manage files across the nodes. The SageFS backend exists on three clusters of machines on the Savi research network. One cluster is in Victoria, one in Carlton, and the third in Toronto.  I used two tools to actually perform sequence alignment, Bowtie2 and MUMmer. Bowtie2 is built to align sequences to a reference genome very quickly using a genome index \cite{langmead2012fast}. Bowtie2 expects the reference genome to be in a binary index format that it can understand, so for these experiments I used the Bowtie2 index of the human genome built from the hg19 build of the human reference genome available on the Bowtie2 sourceforge site. MUMmer is built to align large sequences to each other and simply takes a reference genome and sequences to align as basic arguments \cite{delcher1999alignment}. The reference human genome was the GRCh38 build the NCBI database \cite{geer2009ncbi}. Where as Bowtie2 could align to the entire set of human chromosomes at the same time, MUMmer was run differently in that each viral sequence was run against each human chromosome separately. The viral genomes also came from the NCBI database.